docs/specs/3-13-2023-commands-fit-n-finsh.md

```diff
@@ -0,0 +1,71 @@
+# 3/13/2023 - Commands Fit-and-Finish Proposal
+
+## Design
+
+Commands that interact with the core development workflow will be verb-based. This feels natural to the user as the user is building an application, provisioning infrastructure, and deploying their application to Azure -- which are all actions.
+
+For commands that are of configuration and management in-nature, it is fine to add a noun for a subcommand. For example, `azd env <action>` and `azd config <action>`.
+
+This means that the overall proposed design matches with what `azd` currently offers, and no changes need to occur.
+
+## `azd` usage output
+
+To help introduce user to the command layout, I propose slight changes to the current `azd` usage output (with UX input needed), that looks like the following:
+
+```
+Commands
+  Develop an app
+    init     	: Initialize a new application.
+    provision	: Provision the Azure resources for an application.
+    deploy   	: Deploy the application's code to Azure.
+    up       	: Initialize application, provision Azure resources, and deploy your project with a single command.
+
+  Monitor, test, and release your app
+    monitor  	: Monitor a deployed application.
+    pipeline 	: Manage and configure your deployment pipelines.
+
+  Authenticate with Azure
+    login    	: Log in to Azure.
+    logout   	: Log out of Azure.
+
+  Miscellaneous commands
+    config   	: Manage azd configurations (ex: default Azure subscription, location).
+    down     	: Delete Azure resources for an application.
+    env      	: Manage environments.
+    restore  	: Restore application dependencies.
+    template 	: Find and view template details.
+```
+
+The commands are grouped by their scenarios, with most important scenarios at the top. It is not in alphabetical order, rather the order the user is expected to run the commands. If we added `package` for example, would be after `init`.
```

Comment:
Login isn't required until provision correct?

Reply:
![image](https://user-images.githubusercontent.com/8949415/225048954-3704b02d-e777-4712-bd66-0e4dade298b8.png)
IMO, not strapping login to the front of our commands introduces a number of flows with papercuts.

Reply:
The way I think about is that the user wants to deploy their app to Azure. `login` is simply a necessary requirement as part of this, but it is not the user scenario. The progressive disclosure is exposed in the command error, and I think that is perfectly reasonable.

Reply:
Should we have couple of behaviors based on how one uses azd?
- running a workflow command i.e., azd up. Since being a workflow. I think people will expect the login to happen first or do first thing and let the subsequent commands run. Irrespective of whether the ops is local vs azure.
- running interactively via coarse commands i.e., azd provision, deploy etc. interactively. I think people should be fine here to be forced for a login when needed during provision, deploy.

Reply:
I worry a little about the `login` behavior we might force the user in. If I run `azd init` in a devcontainer with no output, how does `azd` correctly detect that there isn't a browser, and doesn't attempt to incorrectly open and wait for one? When `auth` fails? Does the user know that `auth` failed, and not the command? If it is an authorization failure, is it clear that it came from the command?